addons/stock/models/stock_move.py

```diff
@@ -1231,17 +1249,20 @@ def _recompute_state(self):
                 else:
                     move.state = 'confirmed'
 
-    def _get_upstream_documents_and_responsibles(self, visited):
+    def _get_upstream_documents_and_responsibles(self, visited, log_activity=None):
         if self.move_orig_ids and any(m.state not in ('done', 'cancel') for m in self.move_orig_ids):
             result = set()
             visited |= self
             for move in self.move_orig_ids:
                 if move.state not in ('done', 'cancel'):
-                    for document, responsible, visited in move._get_upstream_documents_and_responsibles(visited):
-                        result.add((document, responsible, visited))
+                    if not move.previous_move_propagate:
+                        log_activity = False
```

Comment:
it doesn't seem clean to me, why should _get_upstream_documents_and_responsibles be ware of the previous_move_propagate?

Reply:
when we get the documents at that time only system will get to know that it has to generate activity on the record or log a note on the record. If move is having an option to cancel the origin then, we must have to log a note instead of an activity, as a system can make changes to the record.